### PR TITLE
web: Add a button to enable local metrics

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @landism,

Please review the following commits I made in branch nicks/localmetrics6:

493460b8f4e768260d095eb0d3921df62f45a7c0 (2020-12-02 13:35:08 -0500)
web: Add a button to enable local metrics

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Review by maiamcc — APPROVED** [review]
lgtm! I feel... a little nervous about `manifest.State` and people remembering to check it where appropriate, but that seems like a problem to deal with it if it happens.

**Comment by maiamcc** [comment]
Screen Shot 2020-12-03 at 10 56 44 AM

Also this isn't a great experience (and the panes don't seem to refresh once the pods are up). Maybe just some copy to the effect of "wait until the pods are up, then refresh"?

**Comment by maiamcc** [comment]
One more note from taking this for a spin myself (doesn't need to be addressed here, but wanted to flag it): the "disable metrics" button doesn't tear down the pods, nor does `tilt down`. Eventually there should be a way to tear those down, esp. because I know a lot of our users are CPU-conscious.

File: internal/engine/metrics/mode_controller_test.go [hunk 1]
```diff
@@ -20,6 +22,38 @@ func TestEnableLocalMetrics(t *testing.T) {
 	defer os.Unsetenv("TILT_METRICS")
 
 	f.mc.OnChange(f.ctx, f.st)
+	assert.Equal(t, model.MetricsDefault, f.st.action.Serving.Mode)
+
+	// Insert a K8s resource into the state store, and make sure that
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/engine/metrics/mode_controller_test.go:27]: hmm...that's what the assert 2 lines above this does? were you looking for something different?

File: web/src/MetricsPane.tsx [hunk 2]
```diff
@@ -29,7 +48,62 @@ let MetricsGraph = styled.iframe`
...
+    <MetricsPaneRoot>
+      <MetricsHeader>Metrics</MetricsHeader>
+      <MetricsDescription>
+        Experimental: Enabling this pane deploys a small metrics stack to your
+        cluster that monitors your build performance.
+        <p />
+        These metrics are not sent outside your cluster.
+        <p />
+        We'd love to{" "}
+        <a href="https://docs.tilt.dev/#community">hear from you</a> on your
```
[the remark this answers is not in the document]
Comment (nicks) [reply web/src/MetricsPane.tsx]: ooh, good point!

File: web/src/MetricsPane.tsx [hunk 3]
```diff
@@ -46,7 +120,14 @@ function MetricsPane(props: { pathBuilder: PathBuilder; serving: Serving }) {
 
   return (
     <MetricsPaneRoot>
-      <MetricsHeader href={link}>Full Dashboard</MetricsHeader>
+      <MetricsHeader>
+        <a href={link}>Full Dashboard</a>
```
[the remark this answers is not in the document]
Comment (nicks) [reply web/src/MetricsPane.tsx]: done

**Review by nicks — COMMENTED** [review]
re: tilt down, that's actually an old issue and happens in lots of cases! https://github.com/tilt-dev/tilt/issues/3137. I would be in favor of fixing that!

re: pods oh that was just a dumb mistake, i had it on PodReadinessModeIgnore. Fixed!

File: internal/engine/metrics/mode_controller_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (maiamcc) [reply internal/engine/metrics/mode_controller_test.go:27]: ah yup, i just missed it while skimming. (if you want to be extra explicit, you can add a comment or a message to the assert--"Expect metrics mode to be default, i.e. unset, when no k8s resources are present" or something.)

**Pull request merged**